ecs-cli/modules/utils/compose/convert_task_definition_test.go

```diff
@@ -43,152 +44,280 @@ var defaultNetwork = &yaml.Network{
 	containerDef := *taskDefinition.ContainerDefinitions[0]
 
 	assert.Nil(t, containerDef.LinuxParameters.SharedMemorySize, "Expected sharedMemorySize to be null")
 }
 
-func TestConvertToTaskDefinitionWithSharedMemorySize(t *testing.T) {
-	// Realistically, we expect customers to specify sizes larger than the default of 64M
-	expectedMBs := 128
-	shmSize := yaml.MemStringorInt(int64(expectedMBs * miB))
-
-	serviceConfig := &config.ServiceConfig{
-		ShmSize:  shmSize,
-		Networks: &yaml.Networks{Networks: []*yaml.Network{defaultNetwork}},
+func TestConvertToTaskDefinitionWithNoTmpfs(t *testing.T) {
+	containerConfig := &containers.ContainerConfig{
+		Tmpfs: nil,
 	}
 
-	taskDefinition := convertToTaskDefinitionInTest(t, "name", nil, serviceConfig, "", "")
+	taskDefinition := convertToTaskDefinitionInTest(t, nil, containerConfig, "", "")
 	containerDef := *taskDefinition.ContainerDefinitions[0]
 
-	assert.Equal(t, int64(expectedMBs), aws.Int64Value(containerDef.LinuxParameters.SharedMemorySize), "Expected sharedMemorySize to match")
+	assert.Nil(t, containerDef.LinuxParameters.Tmpfs, "Expected Tmpfs to be null")
 }
 
+// TODO add test for nil cap add/cap drop
+
+// Test Launch Types
 func TestConvertToTaskDefinitionLaunchTypeEmpty(t *testing.T) {
-	serviceConfig := serviceConfigWithDefaultNetworks()
+	containerConfig := &containers.ContainerConfig{}
 
-	taskDefinition := convertToTaskDefinitionInTest(t, "name", nil, serviceConfig, "", "")
+	taskDefinition := convertToTaskDefinitionInTest(t, nil, containerConfig, "", "")
 	if len(taskDefinition.RequiresCompatibilities) > 0 {
 		t.Error("Did not expect RequiresCompatibilities to be set")
 	}
 }
 
 func TestConvertToTaskDefinitionLaunchTypeEC2(t *testing.T) {
-	serviceConfig := serviceConfigWithDefaultNetworks()
+	containerConfig := &containers.ContainerConfig{}
 
-	taskDefinition := convertToTaskDefinitionInTest(t, "name", nil, serviceConfig, "", "EC2")
+	taskDefinition := convertToTaskDefinitionInTest(t, nil, containerConfig, "", "EC2")
 	if len(taskDefinition.RequiresCompatibilities) != 1 {
 		t.Error("Expected exactly one required compatibility to be set.")
 	}
 	assert.Equal(t, "EC2", aws.StringValue(taskDefinition.RequiresCompatibilities[0]))
 }
 
 func TestConvertToTaskDefinitionLaunchTypeFargate(t *testing.T) {
-	serviceConfig := serviceConfigWithDefaultNetworks()
+	containerConfig := &containers.ContainerConfig{}
 
-	taskDefinition := convertToTaskDefinitionInTest(t, "name", nil, serviceConfig, "", "FARGATE")
+	taskDefinition := convertToTaskDefinitionInTest(t, nil, containerConfig, "", "FARGATE")
 	if len(taskDefinition.RequiresCompatibilities) != 1 {
 		t.Error("Expected exactly one required compatibility to be set.")
 	}
 	assert.Equal(t, "FARGATE", aws.StringValue(taskDefinition.RequiresCompatibilities[0]))
 }
 
+// Test Conversion with ECS Params
+func testContainerConfigs(names []string) []containers.ContainerConfig {
```

Comment:
since this seems to be returning testable objects (and does not take in a *testing.T or assert anything), could we rename to "getTestContainerConfigs"?